From 2b21f18d3421063c423baa6e0c80b0f1682a5b3a Mon Sep 17 00:00:00 2001
From: Jinping Wang <jinping.wang@amlogic.com>
Date: Tue, 30 Oct 2018 18:51:14 +0800
Subject: [PATCH] cec: cec-framework [1/1]

PD# SWPL-1136

Problem:
initializeCec() take to much time in onStart() of HdmiControlService

Solution:
move initializeCec() to a Handler Thread.

Verify:
test ok using r311

Change-Id: Ieb63949b11260b4cba8b16a26ec92515ca168431
Signed-off-by: Jinping Wang <jinping.wang@amlogic.com>
---
 .../core/java/com/android/server/hdmi/HdmiControlService.java  | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/hdmi/HdmiControlService.java b/services/core/java/com/android/server/hdmi/HdmiControlService.java
index 6018383..599c184 100644
--- a/services/core/java/com/android/server/hdmi/HdmiControlService.java
+++ b/services/core/java/com/android/server/hdmi/HdmiControlService.java
@@ -401,7 +401,13 @@ public class HdmiControlService extends SystemService {
         }
         if (mCecController != null) {
             if (mHdmiControlEnabled) {
-                initializeCec(INITIATED_BY_BOOT_UP);
+                mHandler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        initializeCec(INITIATED_BY_BOOT_UP);
+                        initPortInfo();
+                    }
+                });
             } else {
                 mCecController.setOption(OptionKey.ENABLE_CEC, false);
             }
@@ -417,10 +423,10 @@ public class HdmiControlService extends SystemService {
         }
         mMhlDevices = Collections.emptyList();
 
-        initPortInfo();
         if (mMessageValidator == null) {
             mMessageValidator = new HdmiCecMessageValidator(this);
         }
+
         publishBinderService(Context.HDMI_CONTROL_SERVICE, new BinderService());
 
         if (mCecController != null) {

-- 
2.10.2